course3/homework-string/my_string.cc

```diff
@@ -0,0 +1,64 @@
+#include <iostream>
+#include <cstring>
+#include "my_string.h"
+
+MyString::MyString(){
+	buffer = std::unique_ptr<char[]>(new char[1]);
+	buffer = '\0';
```

Comment:
this assign 0 to the _pointer itself_, essentially making it a null pointer:
```
buffer = '\0';
```
What you wanted to do, is to assign 0 to _the first character of the buffer pointed by the pointer_:
```
buffer[0] = '\0';
```